modules/internal/ChapelArray.chpl

```diff
@@ -1558,37 +1558,63 @@ module ChapelArray {
     /* Return true if this domain is a subset of ``super``. Otherwise
        returns false. */
     proc isSubset(super : domain) {
-      if !isAssociativeDom(this) {
-        if isRectangularDom(this) then
-          compilerError("isSubset not supported on rectangular domains");
-        else if isOpaqueDom(this) then
+      if !(isAssociativeDom(this) || isRectangularDom(this)) {
+        if isOpaqueDom(this) then
           compilerError("isSubset not supported on opaque domains");
         else if isSparseDom(this) then
           compilerError("isSubset not supported on sparse domains");
         else
           compilerError("isSubset not supported on this domain type");
       }
       if super.type != this.type then
-        compilerError("isSubset called with different associative domain types");
+        if isRectangularDom(this) {
+          if super.rank != this.rank then
+            compilerError("rank mismatch in _domain.isSubset()");
```

Comment:
This should be 'domain' instead of '_domain'.

'_domain' is the internal type name, but we refer to it as 'domain' for user-facing stuff.

Reply:
@benharsh, noted. Made the required changes.

Reply:
@benharsh, are any other changes required?